packages/repository/src/repositories/relation.repository.ts

```diff
@@ -26,7 +30,47 @@ export interface HasManyEntityCrudRepository<T extends Entity, ID> {
    * @param options Options for the operation
    * @returns A promise which resolves with the found target instance(s)
    */
-  find(filter?: Filter | undefined, options?: Options): Promise<T[]>;
+  find(filter?: Filter, options?: Options): Promise<T[]>;
+  /**
+   * 
+   * @param entity 
+   * @param options 
+   */
+  update(entity: DataObject<T>, options?: Options): Promise<boolean>;
```

Comment:
I am not a fan of the name `update`, because it's not clear whether it's performing a patch (update of a subset of properties provided in `entity`) or a full replace (setting the target model instance exactly to the property values provided in `entity`).

Let's pick one of the more descriptive names (depending on the actual implementation in juggler): `patch` or `replace`.

Reply:
@bajtos good point, I was following the naming convention in [`legacy-juggler-bridge`](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/repositories/legacy-juggler-bridge.ts#L170).
IMO: 
- It makes sense that the crud methods in `legacy-juggler-bridge.ts` align with the names in lb3 juggler
- Given a new opportunity to build lb4 relation from scratch, relation cruds can follow a new naming conversion, which reflects the http verb better and also distinguish partial update(`patch`) and full update(`replace`) in a more clear way.
- It's a little weird that the Entity crud methods and relation crud methods use different naming conversion...
```ts
// entity crud method
customer.update()
// relation crud method
customer.order.patch()
``` 

I am going to sacrifice point 3 in this PR, name the function as `patch`, because given enough bandwidth IMO we should correct some old naming in juggler, not the other way around.
Let's discuss a better strategy for make the naming conversion more consistent across different repositories. Also we have a story to split the repository interface: https://github.com/strongloop/loopback-next/issues/1356.